build-system/tasks/sweep-experiments/index.js

```diff
@@ -0,0 +1,386 @@
+  )
+    .split('\n')
+    .filter((name) => name.endsWith('.js'));
+
+/**
+ * @param {string} fromHash
+ * @return {!Array<string>}
+ */
+const getModifiedSourceFiles = (fromHash) =>
+  getStdout(`git diff --name-only ${fromHash}..HEAD | grep .js`)
+    .split('\n')
+    .filter((file) => !globalWritablePaths.includes(file));
+
+/**
+ * Runs a jscodeshift transform under this directory.
+ * @param {string} transform
+ * @param {Array<string>=} args
+ * @return {string}
+ */
+const jscodeshift = (transform, args = []) =>
+  getStdout(
+    'npx jscodeshift ' +
+      ` -t ${__dirname}/jscodeshift/${transform} ` +
+      args.join(' ')
+  );
+
+/**
+ * @param {string} id
+ * @return {Array<string>} modified files
+ */
+function removeFromExperimentsConfig(id) {
+  jscodeshift(
+    'remove-experiment-config.js'[
+      (`--experimentId=${id}`, experimentsConfigPath)
+    ]
+  );
+  return [experimentsConfigPath];
+}
+
+/**
+ * @param {!Object<string, *>} config
+ * @param {string} path
+ * @param {string} id
+ * @return {Array<string>} modified files
+ */
+function removeFromJsonConfig(config, path, id) {
+  delete config[id];
+  writeFileSync(path, JSON.stringify(config, null, 2) + '\n');
+  return [path];
+}
+
+/**
+ * @param {string} id
+ * @param {number} percentage
+ * @return {Array<string>} modified files
+ */
+function removeFromRuntimeSource(id, percentage) {
+  const possiblyModifiedSourceFiles = filesContainingString(id);
+  if (possiblyModifiedSourceFiles.length > 0) {
+    jscodeshift('remove-experiment-runtime.js', [
+      `--isExperimentOnLaunched=${percentage}`,
+      `--isExperimentOnExperiment=${id}`,
+      ...possiblyModifiedSourceFiles,
+    ]);
+  }
+  return possiblyModifiedSourceFiles;
+}
+
+/**
+ * @param {string} str
+ * @return {string}
+ */
+const cmdEscape = (str) => str.replace(/["`]/g, (c) => `\\${c}`);
+
+/**
+ * @param {string} id
+ * @param {*} workItem
+ * @param {!Array<string>} modified
+ * @return {?string}
+ */
+function gitCommitSingleExperiment(
+  id,
+  {previousHistory, percentage},
+  modified
+) {
+  exec(`git add ${modified.join(' ')}`);
+  const commitMessage =
+    `${readableRemovalId(id, {previousHistory, percentage})}\n\n` +
+    `${prodConfigPath.split('/').pop()} history:\n\n` +
+    previousHistory
+      .map(
+        ({hash, authorDate, subject}) =>
+          ` - ${hash} - ${authorDate}\n   ${subject}\n`
+      )
+      .join('\n');
+  return getStdout(`git commit -m "${cmdEscape(commitMessage)}"`);
+}
+
+function readableRemovalId(id, {percentage, previousHistory}) {
+  const {knownPr, authorDate, hash} = previousHistory[0];
+  return (
+    `(${truncateYyyyMmDd(authorDate)}, ${knownPr || hash}) ` +
+    `\`${id}\`: ${percentage}`
+  );
+}
+
+/**
+ * @param {!Array<string>=} extraFiles
+ * @return {*}
+ */
+const lintFix = (extraFiles = []) =>
+  getOutput(
+    `gulp lint --fix --files="${[experimentsConfigPath]
+      .concat(extraFiles)
+      .join(',')}"`
+  );
```

Comment:
This is rather slow and I'm not sure why. Maybe we can plug into its internal utilities?